lib/media/segment_prefetch.js

```diff
@@ -87,15 +81,17 @@ shaka.media.SegmentPrefetch = class {
       shaka.log.debug(logPrefix, 'missing segmentIndex');
       return;
     }
-    const maxTime = Math.max(currTime, this.prefetchPosTime_);
-    const iterator = this.stream_.segmentIndex.getIteratorForTime(
-        maxTime, /* allowNonIndepedent= */ true);
-    if (!iterator) {
+    if (!this.iterator_) {
+      this.iterator_ = this.stream_.segmentIndex.getIteratorForTime(
```

Comment:
Since this part of the code is being touched, I see that the `setReverse` function of the iterator is not being implemented, please implement it in the scope of this PR.

Reply:
Done!